src/components2/MapLegend/MapLegend.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { breakpoints } from '~/styles/utils';
+
+import LegendExpanded from './LegendExpanded';
+
+interface Props {
+  isPopupVisible: boolean;
+  isDetailOpen: boolean;
+}
+
+const Wrapper = styled.div<Props>`
+  display: ${({ isPopupVisible, isDetailOpen }) =>
+    !isPopupVisible && !isDetailOpen ? 'block' : 'none'};
+
+  @media (min-width: ${breakpoints.m}px) {
+    display: ${({ isDetailOpen }) => (!isDetailOpen ? 'block' : 'none')};
+  }
```

Comment:
The "none" parts are IMO a guard in the component, are they not? Like 

```jsx
const MapLegend = (props: Props) => {
  if(!visible && !open) return;

  return (<Wrapper>…</Wrapper>)
}
```

Which makes this all go away.